hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/metadata/TestHoodieBackedMetadata.java

```diff
@@ -120,46 +120,63 @@ public void testDefaultNoMetadataTable() throws Exception {
     assertThrows(TableNotFoundException.class, () -> HoodieTableMetaClient.builder().setConf(hadoopConf).setBasePath(metadataTableBasePath).build());
 
     // Metadata table is not created if disabled by config
+    String firstCommitTime = HoodieActiveTimeline.createNewInstantTime();
     try (SparkRDDWriteClient client = new SparkRDDWriteClient(engineContext, getWriteConfig(true, false))) {
-      client.startCommitWithTime("001");
-      client.insert(jsc.emptyRDD(), "001");
+      client.startCommitWithTime(firstCommitTime);
+      client.insert(jsc.parallelize(dataGen.generateInserts(firstCommitTime, 5)), firstCommitTime);
```

Comment:
would we remove calling `syncTableMetadata()` in `SparkRDDWriteClient#preWrite`? since it will do nothing as there are in progress instant always.

Reply:
In the bootstrap path this will do nothing. But in the normal path (post bootstrap), this path also syncs the metadata table.

The metadata table sync is indepdendant of the dataset commits. So its possible that the dataset commit may have completed and metadata table sync failed (due to some error or crash) in postWrite. So calling the sync again in preWrite ensures we sync again.

@vinothchandar Do you think the sync in preWrite can be removed as an optimization?

Reply:
As @prashantwason mentioned, its actually kind of important that we do that in preWrite() so we bring the metadata table in sync with the timeline, before writes happen. So not sure if it can be removed. 

We are moving towards a synchronous design anyway, for updating metadata, so lets may be revisit in that context?